spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployerTests.java

```diff
@@ -499,40 +501,67 @@ public void testStateOfDefinedUndeployedStream() {
 
 	@Test
 	@SuppressWarnings("unchecked")
-	public void testUndeploySkippedForUndefinedStream() {
+	public void testUndeployPackageAndReleaseExistAllGood() {
 		AppRegistryService appRegistryService = mock(AppRegistryService.class);
 		SkipperClient skipperClient = mock(SkipperClient.class);
 		StreamDefinitionRepository streamDefinitionRepository = mock(StreamDefinitionRepository.class);
-
 		SkipperStreamDeployer skipperStreamDeployer = new SkipperStreamDeployer(skipperClient,
 				streamDefinitionRepository, appRegistryService, mock(ForkJoinPool.class), new DefaultStreamDefinitionService());
-
 		StreamDefinition streamDefinition = new StreamDefinition("foo", "foo|bar");
-
-		when(skipperClient.search(eq(streamDefinition.getName()), eq(false))).thenReturn(new ArrayList<>());
+		PackageMetadata packageMetadata = new PackageMetadata();
```

Comment:
These tests are too mocked out - only really verifying that the impl I have added is in fact what gets called. Ultimately we need to do the following in a full integration test:

- Create stream `foo`
- Create stream `foobar`
- Delete foo package data (for real or mocked out)
- Delete stream `foo`
- Verify stream `foo` deleted
- Create stream `foo` 
- Verify it creates stream w/o barfing about "release already exists".

Reply:
We need a test that launches rabbit, skipper and database using containers and then using streamservice to deploy tests and jdbctemplate to delete the entries. Then service to perform delete

Reply:
Exactly.